app/components/npq_separation/admin/secondary_navigation_component.rb

```diff
@@ -0,0 +1,107 @@
+module NpqSeparation
+  module Admin
+    class SecondaryNavigationComponent < ViewComponent::Base
+      attr_accessor :current_path
+
+      Section = Struct.new(:name, :path, :prefix, :nodes, keyword_init: true)
+      Node = Struct.new(:name, :path, :prefix, keyword_init: true)
+
+      def initialize(current_path)
+        @current_path = current_path
+      end
+
+      def sections
+        structure.fetch(current_section)
+      end
+
+      def render?
+        current_section.present?
+      end
+
+      def secondary_navigation_link(section)
+        link_to(
+          section.name,
+          section.path,
+          class: "x-govuk-sub-navigation__link",
+          aria: { current: current?(section.prefix) },
+        )
+      end
+
+      def secondary_navigation_item_link(node)
+        link_to(
+          node.name,
+          node.path,
+          class: "x-govuk-sub-navigation__link",
+          aria: { current: current?(node.prefix) },
+        )
+      end
+
+    private
+
+      def current?(prefix)
+        # return nil instead of false so Rails' link helper drops the
+        # attribute rather than setting "current='false'"
+        current_path.start_with?(prefix) || nil
+      end
+
+      def current_section
+        all_sections = [
+          Node.new(name: "Applications", prefix: "/npq-separation/admin/applications"),
+          Node.new(name: "Finance", prefix: "/npq-separation/admin/finance"),
+          Node.new(name: "Schools", prefix: "/npq-separation/admin/schools"),
+          Node.new(name: "Lead providers", prefix: "/npq-separation/admin/lead-providers"),
+          Node.new(name: "Settings", prefix: "/npq-separation/admin/settings"),
+
+          # Dashboard is a special case as it's the admin 'landing page' so has the
+          # '/admin' path as well as '/admin/dashboards/summary', so we'll match it
+          # last
+          Node.new(name: "Dashboard", prefix: %r{/npq-separation/(admin|admin/dashboard)}),
+        ]
+
+        matching_section = all_sections.find { |ps| current_path.start_with?(ps.prefix) }
+
+        if matching_section.nil?
+          Rails.logger.error("No matching admin section for '#{current_path}'")
+
+          return
+        end
+
+        matching_section.name
+      end
+
+      def structure
```

Comment:
I hardcoded the structure here for now, we could write some extra specs that ensure the names/paths/prefixes are correct if we're happy with the approach.